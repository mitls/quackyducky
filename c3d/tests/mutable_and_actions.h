#include <stdint.h>

typedef struct
  [[everparse::process(1)]]
  [[everparse::entrypoint]]
  [[everparse::mutable_parameter(UINT32 *x)]]
  [[everparse::mutable_parameter(UINT32 *y)]]
_Pair
{
   UINT32 first
       [[everparse::on_success(^{*x = first;
                                 return true; })]];
   UINT32 second
       [[everparse::on_success(^{*y = second;
                                 return true; })]];
} Pair;
