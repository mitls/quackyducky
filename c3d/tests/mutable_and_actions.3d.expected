entrypoint
typedef struct _Pair (mutable uint32_t * x, mutable uint32_t * y) { 
  uint32_t first
    {:on-success 
{
    *x = first;
    return 1;
}

    };
  uint32_t second
    {:on-success 
{
    *y = second;
    return 1;
}

    };
} Pair, *PPair;

